sktime/transformations/series/boxcox.py

```diff
@@ -528,7 +528,8 @@ def _guerrero(x, sp, bounds=None):
 
     if sp is None or not is_int(sp) or sp < 2:
         raise ValueError(
-            "Guerrero method requires an integer seasonal periodicity (sp) value >= 2."
+            "In BoxCoxTransformer, method='guerrero' requires an integer seasonal "
+            f"periodicity (sp) value >= 2, but found {sp}"
```

Comment:
```suggestion
            f"periodicity (sp) value >= 2, but found {sp=}"
```

Completely non-blocking suggestion, looks a bit nicer like `found sp=1` instead of `found 1`.

Reply:
but that would be `sp={sp}`, not the suggested update - so I will not accept the update but fix manually

Reply:
@fkiraly this is what I meant.

![image](https://github.com/sktime/sktime/assets/39331844/f3fcfd19-4218-42d6-9ccf-5ed604ca01b3)

Ref. https://docs.python.org/3/whatsnew/3.8.html#f-strings-support-for-self-documenting-expressions-and-debugging

Reply:
oh, I did not know that worked! This is neat, thanks for the learning moment.